pkg/services/pluginsintegration/pluginclient/pluginclientimpl/facade.go

```diff
@@ -0,0 +1,158 @@
+package pluginclientimpl
+
+import (
+	"context"
+	"errors"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/services/contexthandler"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginclient"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/plugincontext"
+	"github.com/grafana/grafana/pkg/services/validations"
+)
+
+type Facade struct {
```

Comment:
🤔 -- rather than having a facade let you talk to any plugin, can the facade be a helper that injects config into a single plugin?

eg, rather than adding a reference to everything, you use a reference to get a facade, then call it (without plugin context)

Reply:
I guess similar question/comment as earlier comment. Sure that could work, but I would probably still suggest separate client/request structs to streamline the consumer API.